.github/workflows/tests.yml

```diff
@@ -149,69 +163,93 @@ jobs:
                   continue
                 elif [[ $CODE -ne 0 ]]; then
                   echo "::error::Script skip-$TEST_SCRIPT failed to run $CODE!"
+                  FAILED+=( "$SLUG" )
                   EXIT=1
                   continue
                 fi
               fi
 
-              echo "::group::Running tests for $SLUG"
-
-              # Composer install, if appropriate. Note setup-wordpress-env.sh did it already for plugins.
-              NEED_COMPOSER=false
-              if [[ "${SLUG%%/*}" != "plugins" && ( "$TEST_SCRIPT" == "test-php" || "$TEST_SCRIPT" == "test-coverage" ) ]]; then
-                NEED_COMPOSER=true
-                if [[ "$TEST_SCRIPT" == "test-coverage" ]] &&
-                  ! jq -e '.scripts["test-php"]' "$DIR/composer.json" &>/dev/null
-                then
-                  echo "Skipping composer install, assuming test-coverage is only JS because the project has no test-php."
-                  NEED_COMPOSER=false
-                fi
-              fi
-              if $NEED_COMPOSER; then
-                if [[ ! -f "$DIR/composer.lock" ]]; then
-                  echo 'No composer.lock, running `composer update`'
-                  composer --working-dir="$DIR" update
-                elif composer --working-dir="$DIR" check-platform-reqs --lock; then
-                  echo 'Platform reqs pass, running `composer install`'
-                  composer --working-dir="$DIR" install
-                else
-                  echo 'Platform reqs failed, running `composer update`'
-                  composer --working-dir="$DIR" update
+              echo "Running tests for $SLUG"
+              {
+                # Composer install, if appropriate. Note setup-wordpress-env.sh did it already for plugins.
+                NEED_COMPOSER=false
+                if [[ "${SLUG%%/*}" != "plugins" && ( "$TEST_SCRIPT" == "test-php" || "$TEST_SCRIPT" == "test-coverage" ) ]]; then
+                  NEED_COMPOSER=true
+                  if [[ "$TEST_SCRIPT" == "test-coverage" ]] &&
+                    ! jq -e '.scripts["test-php"]' "$DIR/composer.json" &>/dev/null
+                  then
+                    echo "Skipping composer install, assuming test-coverage is only JS because the project has no test-php."
+                    NEED_COMPOSER=false
+                  fi
                 fi
+                if $NEED_COMPOSER; then
```

Comment:
Since you're already messing with the indent anyway, you could consider collapsing this with the above iff so you don't need `NEED_COMPOSER`.

Reply:
dde1131379